deepchecks/recommender/checks/cold_start_detection.py

```diff
@@ -0,0 +1,101 @@
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+from typing import List, Union
+import pandas as pd
+import numpy as np
+from PyNomaly import loop
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.errors import (DeepchecksProcessError, DeepchecksTimeoutError, DeepchecksValueError,
+                                    NotEnoughSamplesError)
+from deepchecks.tabular import  SingleDatasetCheck
+from deepchecks.recommender import  Context, InteractionDataset
+from deepchecks.utils.dataframes import select_from_dataframe
+from deepchecks.utils.strings import  format_percent
+from deepchecks.utils.typing import Hashable
+import plotly.express as px
+
+__all__ = ['ColdStartDetection']
+
+
+
+class ColdStartDetection(SingleDatasetCheck):
+    """ Retrieve cold start users and items. \
+        Cold start entity refers to new entity who have limited or no historical data or interactions within a system or platform. \
+        They have not yet established a significant user/item history that can be leveraged for personalized recommendations or targeted actions. \
+        For the model, there can be some consequences fromincreaseduncertainty to the difficulty in personalization. \
+        From the business side, the consequences can be reduced user engagement,missed opportunities, and limited user understanding.
+        The model is trained on the remaining historical data for each user and then evaluated on the held-out interactions. \
+        
+    Parameters
+    ----------
+    columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to check, if none are given checks all columns except ignored ones.
+    ignore_columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to ignore, if none given checks based on columns variable
+    n_samples : int , default: 5_000
+        number of samples to use for this check.
+    n_to_show : int , default: 5
+        number of data elements with the highest outlier score to show (out of sample).
+    random_state : int, default: 42
+        random seed for all check internals.
+    """
+
+    def __init__(
+            self,
+            columns: Union[Hashable, List[Hashable], None] = None,
+            ignore_columns: Union[Hashable, List[Hashable], None] = None,
+            n_samples: int = 5_000,
+            n_to_show: int = 10,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.random_state = random_state
+
+    def run_logic(self, context: Context, dataset_kind : InteractionDataset) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
```

Comment:
Can you please refresh my memory - how does the context know to pass "dataset_kind=interaction' for this check?